File: web/filing/models/tax-credits/tax-paid-self-model.ts
/**
 * Tax Paid Self Model
 * Advance Tax & Self Assessment Tax
 */
export interface TaxPaidSelfModel {
  taxPaidId?: number | null
  filingId: number | null
  challanNumber?: string | null
  bsrCode?: string | null
  dateOfPayment?: any | null
  taxPaidAmount?: number | null
  taxPaidDate?: any | null
  taxType?: string | null
}
export interface TaxPaidSelfResponse {
  data: TaxPaidSelfModel
  message?: string
}

export interface TaxPaidSelfListResponse {
  data: TaxPaidSelfModel[]
  message?: string
}
export interface TaxPaymentWidgetProps {
  isOpen: boolean
  onClose: () => void
  filingId: number | null
  entityId?: number | string
  initialData?: Record<string, any> | null
  onSuccess?: (action: 'created' | 'updated', data: any) => void
  errorField?: string | null
  errorMessage?: string | null  // Error message to display for the field
  fieldErrors?: Record<string, string> | null  // Multiple field errors: { fieldName: errorMessage }
}

File: api/filing/models/tax_paid_self_model.py
"""Tax Paid Self model (Advance Tax / Self-Assessment Tax)."""
from typing import Any
from .api_base_model import ApiBaseModel


class TaxPaidSelfModel(ApiBaseModel):
    tax_paid_id: int | None = None
    filing_id: int | None = None
    challan_number: str | None = None
    bsr_code: str | None = None
    date_of_payment: Any | None = None
    tax_paid_amount: float | str | int | None = None
    tax_paid_date: Any | None = None
    tax_type: str | None = None
